storage/fallocate_darwin.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2018 The Chubao Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package storage
+
+import (
+	"syscall"
+
+	"github.com/chubaofs/chubaofs/util/log"
+)
+
+func fallocate(fd int, mode uint32, off int64, len int64) (err error) {
+	// system call 'fallocate' is not supported in Darwin(Apple MacOS).
+	log.LogWarnf("fallocate: not supported in Darwin(Apple MacOS) operating system")
```

Comment:
[go-fallocate](https://github.com/detailyang/go-fallocate/blob/master/fallocate_darwin.go#L9)  implement syscall "fallocate" with different os. How about import it here？

Reply:
Thank you for your suggestion.
We will follow up and test your proposal.

Reply:
> [go-fallocate](https://github.com/detailyang/go-fallocate/blob/master/fallocate_darwin.go#L9) implement syscall "fallocate" with different os. How about import it here？

Thanks in advance for your suggestions.
But your proposal does not match our requirement.

We use `fallocate` system call with **hole punching** flag to release disk space used by unused tiny files. 

The `go-fallocate` can not make possible in hole punching.

More detail about our storage engine  see our paper [CFS: A Distributed File System for Large Scale Container Platforms](https://arxiv.org/abs/1911.03001).

Reply:
Yep, I see. Will do it later.